velox/expression/CastExpr-inl.h

```diff
@@ -340,7 +340,7 @@ void CastExpr::applyCastKernel(
             makeErrorMessage(*input, row, result->type(), details);
         context.setStatus(row, Status::UserError("{}", errorDetails));
       } else {
-        context.setStatus(row, Status::UserError("{}", details));
+        context.setStatus(row, Status::UserError());
```

Comment:
could encapsulate some of the logic in a macro? Maybe something like:

```
VELOX_CAPTURE_STATUS(context, StatusCode::kUserError, "my error msg: {}", details);
```

that expands to something similar to this branch?

Reply:
'context' is specific to expression evaluation, hence, the macro defined like above would be applicable only within expression eval.